app/src/main/java/org/oppia/android/app/story/StoryFragmentPresenter.kt

```diff
@@ -142,6 +150,40 @@ class StoryFragmentPresenter @Inject constructor(
             ).parseOppiaHtml(
               storyItemViewModel.summary, binding.chapterSummary
             )
+          if (storyItemViewModel.chapterSummary.chapterPlayState
+            == ChapterPlayState.NOT_PLAYABLE_MISSING_PREREQUISITES
+          ) {
+            val missingPrerequisiteSummary = fragment.getString(
+              R.string.chapter_prerequisite_title_label,
+              storyItemViewModel.index.toString(),
+              storyItemViewModel.missingPrerequisiteChapter.name
+            )
+            val chapterLockedSpannable = SpannableString(missingPrerequisiteSummary)
+            val clickableSpan = object : ClickableSpan() {
+              override fun onClick(widget: View) {
+                smoothScrollToPosition(storyItemViewModel.index - 1)
+              }
+
+              override fun updateDrawState(ds: TextPaint) {
+                super.updateDrawState(ds)
+                ds.isUnderlineText = false
+              }
+            }
+            chapterLockedSpannable.setSpan(
+              clickableSpan,
+              9,
```

Comment:
Please mention inline comment
`/* start= */ 9,`
Also, do mention why are you subtracting `9` only.

Reply:
Sure.
You see we are showing a particular string when the chapter is locked. The string is `Complete Chapter %s: %s to unlock this chapter.` Here the only part that is to be made a clickable span is `%s: %s` which is replaced on runtime by the arguments provided in `getString()` while initializing `missingPrerequisiteSummary`. Thus, the rest part of the string remains constant. If we count the number of character from start to the first placeholder `%s`, we will find that to be **9**. Hence, I put the `start` range of the `clickableSpan` as **9**.
Hope I made it clear. Please share your thoughts. Thanks.

Also, I have added the inline comments. PTAL.

Reply:
Thanks for explanation but having something in code which describes this will make sure that others can understand this.

Suggestion:
Create constants like
`private val const LOCKED_CARD_PREFIX_LENGTH = 9`
`private val const LOCKED_CARD_SUFFIX_LENGTH = 24`

and use it everywhere.

Reply:
Sure. Thanks for the advice.

Reply:
Done.